### PR TITLE
Test and validator for endpoint put post blog61

### PR DESCRIPTION
**Pull request by mirado447**

### CODE REVIEW

File: src/main/java/com/blogify/blogapi/model/validator/PostValidator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      throw new BadRequestException("Post_id is mandatory");
+    }
+    if(!violations.isEmpty()){
+      String constraintMessages =
+         violations.stream()
+             .map(ConstraintViolation::getMessage)
+             .collect(Collectors.joining(". "));
+      throw new BadRequestException(constraintMessages);
+    }
+  }
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/model/validator/PostValidator.java]: non non comme j'ai dit utilice cette methode:
@Override
  public void accept(Fee fee) {
    Set<String> violationMessages = new HashSet<>();
    if (fee.getStudent() == null) {
      violationMessages.add("Student is mandatory");
    }
    if (fee.getDueDatetime() == null) {
      violationMessages.add("Due datetime is mandatory");
    }
    if (fee.getTotalAmount() < 0) {
      violationMessages.add("Total amount must be positive");
    }
    if (!violationMessages.isEmpty()) {
      String formattedViolationMessages =
          violationMessages.stream().map(String::toString).collect(Collectors.joining(". "));
      throw new BadRequestException(formattedViolationMessages);
    }
  }
  
  mais pour la version post

File: src/main/java/com/blogify/blogapi/service/PostService.java [hunk 2]
```diff
@@ -46,6 +48,7 @@ public Post savePost(Post post, String postId) {
     Optional<Post> optionalPost = postRepository.findById(postId);
     if (!optionalPost.isEmpty()) {
       Post postToUpdate = optionalPost.get();
+      postValidator.accept(post);
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/service/PostService.java]: not pa ici, juste apres le "public Post savePost(Post post, String postId) {"

File: src/main/java/com/blogify/blogapi/model/validator/PostValidator.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+public class PostValidator implements Consumer<Post> {
+  private  final Validator validator;
+
+  public void  accept(List<Post> posts){posts.forEach(this);}
+  @Override
+  public void accept(Post post){
+    Set<ConstraintViolation<Post>> violations = validator.validate(post);
+    if(post.getId() == null){
+      throw new BadRequestException("Post_id is mandatory");
+    }
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/model/validator/PostValidator.java]: ajoute aussi
user is mandatory

File: src/test/java/com/blogify/blogapi/integration/PostIT.java [hunk 4]
```diff
@@ -151,6 +152,21 @@ void client_write_ok() throws ApiException {
 
     assertEquals(3, allPostsAfterCreate.size());
   }
+  @Test
+  @DirtiesContext
+  void client_update_ko() throws ApiException{
+    ApiClient client1 = apiClient(CLIENT1_TOKEN);
+    PostingApi api = new PostingApi(client1);
+    Post post = api.getPostById(POST1_ID);
+    Post postUpdate = post.id(null);
```
Comment (heiherilala) [inline src/test/java/com/blogify/blogapi/integration/PostIT.java]: ajoute aussi mour user is mandatory

regarde HEI admin pour example

**Review by heiherilala — APPROVED** [review]

File: src/main/java/com/blogify/blogapi/model/validator/PostRestValidator.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+@AllArgsConstructor
+public class PostRestValidator implements Consumer<com.blogify.blogapi.endpoint.rest.model.Post> {
+    private  final Validator validator;
+
+    public void  accept(List<com.blogify.blogapi.endpoint.rest.model.Post> posts){posts.forEach(this);}
+    @Override
+    public void accept(Post post){
+        Set<String> violationMessages = new HashSet<>();
+        if (post.getAuthorId() == null){
+            violationMessages.add("User is mandatory");
```
Comment (fenohasinalala) [inline src/main/java/com/blogify/blogapi/model/validator/PostRestValidator.java]: change to author_id is mandatory

File: src/test/java/com/blogify/blogapi/integration/PostIT.java [hunk 6]
```diff
@@ -151,6 +152,25 @@ void client_write_ok() throws ApiException {
...
+    ApiException exception1 = assertThrows(ApiException.class,
+            () -> api.crupdatePostById(POST1_ID,postUpdate1));
+    ApiException exception2 = assertThrows(ApiException.class,
+            () -> api.crupdatePostById(POST1_ID,postUpdate2));
+
+    String exceptionMessage1 = exception1.getMessage();
+    String exceptionMessage2 = exception2.getMessage();
+
+    assertTrue(exceptionMessage1.contains("Post_id is mandatory"));
+    assertTrue(exceptionMessage2.contains("User is mandatory"));
```
Comment (fenohasinalala) [inline src/test/java/com/blogify/blogapi/integration/PostIT.java]: change to author_id is mandatory

**Pull request merged**